test/suite/test_oligarch15.py

```diff
@@ -0,0 +1,123 @@
+
+    conn_config = 'oligarch_log=(enabled),statistics=(all),statistics_log=(wait=1,json=true,on_close=true),' \
+                + 'disaggregated=(stable_prefix=.,page_log=palm,role="follower"),'
+
+    create_session_config = 'key_format=S,value_format=S'
+
+    oligarch_uris = ["oligarch:test_oligarch15a", "oligarch:test_oligarch15b"]
+    # FIXME-SLS-555 Add the "table:" URI to the list once it is supported
+    # other_uris = ["file:test_oligarch15c", "table:test_oligarch15d"]
+    other_uris = ["file:test_oligarch15c"]
+
+    disagg_storages = gen_disagg_storages('test_oligarch15', disagg_only = True)
+    scenarios = make_scenarios(disagg_storages)
+
+    # Load the page log extension, which has object storage support
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        DisaggConfigMixin.conn_extensions(self, extlist)
+
+    # Custom test case setup
+    def early_setup(self):
+        os.mkdir('kv_home')
+
+    # Test starting without local files.
+    def test_oligarch14(self):
+        # The node started as a follower, so step it up as the leader
+        self.conn.reconfigure('disaggregated=(role="leader")')
+
+        # Create tables
+        for uri in self.oligarch_uris + self.other_uris:
+            cfg = self.create_session_config
+            if not uri.startswith('oligarch'):
+                cfg += ',block_manager=disagg,oligarch_log=(enabled=false),log=(enabled=false)'
+            self.session.create(uri, cfg)
+
+        # Put data to tables
+        value_prefix = 'aaa'
+        for uri in self.oligarch_uris + self.other_uris:
+            cursor = self.session.open_cursor(uri, None, None)
+            for i in range(self.nitems):
+                cursor[str(i)] = value_prefix + str(i)
+                if i % 250 == 0:
+                    time.sleep(1)
+            cursor.close()
+
+        time.sleep(1)
+        self.session.checkpoint()
+        time.sleep(1)
+        checkpoint_id = self.disagg_get_complete_checkpoint()
+
+        # Clean up
+        self.close_conn()
+
+        # Remove local files
+        for f in os.listdir():
+            if os.path.isdir(f):
+                continue
+            if f.startswith('WiredTiger') or f.startswith('test_'):
+                os.remove(f)
+
+        # Reopen the connection
+        self.open_conn()
+
+        # Recreate oligarch tables
+        # FIXME-SLS-496 Remove this after we can create oligarch table metadata automatically
+        for uri in self.oligarch_uris:
+            cfg = self.create_session_config
+            self.session.create(uri, cfg)
+
+        # Pick up the checkpoint
+        self.conn.reconfigure(f'disaggregated=(checkpoint_id={checkpoint_id})')
+
+        # Become the leader (skip a few extra checkpoint IDs just in case)
+        self.conn.reconfigure(f'disaggregated=(role="leader",next_checkpoint_id={checkpoint_id+2})')
+
+        # Check tables in the follower
```

Comment:
We become the leader, but then the next comment says we are checking tables in the _follower_.
(Is there a reason we had to become the leader to check the data?)